File: solutions/go/resistor-color-duo/1/resistor_color_duo.go
package resistorcolorduo

// Value should return the resistance value of a resistor with a given colors.
func Value(colors []string) int {
	colorMap := map[string]int{
		"black":  0,
		"brown":  1,
		"red":    2,
		"orange": 3,
		"yellow": 4,
		"green":  5,
		"blue":   6,
		"violet": 7,
		"grey":   8,
		"white":  9,
	}

	if len(colors) < 2 {
		return 0 // not enough colors to form a value
	}

	first := colorMap[colors[0]]
	second := colorMap[colors[1]]

	return first*10 + second
}

File: solutions/typescript/resistor-color-duo/1/resistor-color-duo.ts
const COLORS = [
  "black",
  "brown",
  "red",
  "orange",
  "yellow",
  "green",
  "blue",
  "violet",
  "grey",
  "white",
];


export function decodedValue(colors:string[]) : number{
  if(!colors || colors.length < 2){
    throw new Error("At least two colors are required");
  }

  const first = COLORS.indexOf(colors[0].toLowerCase())
  const second = COLORS.indexOf(colors[1].toLowerCase())


  if(first === -1 || second === -1){
    throw new Error("Invalid color provided");
  }

  return first * 10 + second;
}
